src/store/singing.ts

```diff
@@ -886,104 +888,192 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
    */
   RENDER: {
     async action({ state, getters, commit, dispatch }) {
-      const searchPhrases = async (notes: Note[]) => {
-        const foundPhrases = new Map<string, Phrase>();
+      const calcPhraseFirstRestDuration = (
+        prevPhraseLastNote: Note | undefined,
+        phraseFirstNote: Note,
+        phraseFirstRestMinDurationSeconds: number,
+        tempos: Tempo[],
+        tpqn: number,
+      ) => {
```

Comment:
パラメータ多くなってきたのでこの辺の値を指定し間違えて発見しづらいバグに繋がったりしやすいかもと感じました！
こんな感じで引数の型指定すると引数名を書く形で関数呼び出しできるので間違いづらくなるかも。
（その代わり引き数の型がややこしくなってしまうのですが。。）

```suggestion
      const calcPhraseFirstRestDuration = ({
        prevPhraseLastNote,
        phraseFirstNote,
        phraseFirstRestMinDurationSeconds,
        tempos,
        tpqn,
      }: {
        prevPhraseLastNote: Note | undefined,
        phraseFirstNote: Note,
        phraseFirstRestMinDurationSeconds: number,
        tempos: Tempo[],
        tpqn: number,
      }) => {
```